mantidproject/mantid
# Enable cross compilation on M1 Macs

**Description of work**

This PR enables cross compilation on macs with an ARM-based M1 chip. This enables M1 macs to compile packages than can run on older Intel based x84-64 chips.

This PR is necessary so that we can upgrade our old Mac hardware that currently is a bottleneck on the CI/CD pipeline.

Having M1 Macs operational also provides opportunity (and security) in the future for packages of mantid to be released targeting ARM-based machines.

**To test:**

- Ensure tests pass on this PR, this demonstrates that developer builds on Windows, Linux and x84-64 macs remain functional given this change.
- Ensure the following package build passes:
  https://builds.mantidproject.org/job/build_packages_from_branch/417/





---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
